### ISSUE TITLE
handle_response is not correct

### ISSUE THREAD

**Issue by free1002** [geeknam/python-gcm#3]
1. When I tried send a message using plaintext_request, a response format was like this :

<< success case >>
<code>
res : id=0:1342406578446357%24fa77d8f9fd7ecd
</code>

<< failure case >>
<code>
res :  Error=InvalidRegistration
</code>

But, current handle_response has a assumption that the type of response is dict. This assumption is not true.
1. When I tried json_request, the response  was like this : ( res_ids has 3 items. First and third reg_ids are wrong register id and second reg_id is right register id)

<code>
{u'failure': 2, u'canonical_ids': 0, u'success': 1, u'multicast_id': 6941921051371432547, u'results': [{u'error': u'InvalidRegistration'}, {u'message_id': u'0:1342406147892805%24fa77d8f9fd7ecd'}, {u'error': u'InvalidRegistration'}]}
</code>

But, in handle_response, 'error' is set like this :
<code>
error = response['results']['error']
</code>

In conclusion, handle_response is not right implementation for both of plaintext_request and json_request.
